src/common/Testimonial/TestimonialModal.jsx

```diff
@@ -0,0 +1,198 @@
+import React, { useEffect, useState } from 'react';
+import Modal from '@mui/joy/Modal';
+import ModalClose from '@mui/joy/ModalClose';
+import Typography from '@mui/joy/Typography';
+import Sheet from '@mui/joy/Sheet';
+import { useUserDisplayName, useUserId } from '@nhost/react';
+import { Box, TextField } from '@mui/material';
+import Button from '@mui/joy/Button';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select from '@mui/material/Select';
+import {
+  fetchAllEvents,
+  insert_testimonial_submission
+} from 'common/services/request/query/fetch-testimonials';
+import { submit } from 'common/services/request';
+import { toast, ToastContainer } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
+export default function TestimonialModal({ isOpen, setIsOpen }) {
+  const userDisplayName = useUserDisplayName();
+  const userId = useUserId();
+  const [testimonialData, setTestimonialData] = useState({
+    quote: '',
+    title: '',
+    event: '',
+    id: userId
+  });
+  const [btnDisabled, setBtnDisabled] = useState(true);
+
+  const [Events, setEvents] = useState([]);
+
+  const fetchTestEvents = async () => {
+    const res = await submit(fetchAllEvents());
+    setEvents(res);
+  };
+
+  useEffect(() => {
+    fetchTestEvents();
+  }, []);
+
+  const updateData = (e) => {
+    const fieldName = e.target.name;
+    setTestimonialData((prev) => ({
+      ...prev,
+      [fieldName]: e.target.value
+    }));
+
+    if (testimonialData.title.length != 0 && testimonialData.quote.length != 0) {
+      setBtnDisabled(false);
+    }
+  };
+
+  const AddTestimonial = async () => {
```

Comment:
function name should start with small case.. please change everywhere applicable.